app/src/main/java/com/infomaniak/drive/ui/LaunchActivity.kt

```diff
@@ -252,16 +245,17 @@ class LaunchActivity : AppCompatActivity() {
     private fun setPublicShareActivityArgs(
         driveId: String,
         publicShareUuid: String,
-        fileId: Int = -1,
+        shareLink: ShareLink? = null,
         isPasswordNeeded: Boolean = false,
         isExpired: Boolean = false,
     ) {
         publicShareActivityExtras = PublicShareActivityArgs(
             driveId = driveId.toInt(),
             publicShareUuid = publicShareUuid,
-            fileId = fileId,
+            fileId = shareLink?.fileId ?: -1,
```

Comment:
Add a sentry if needed

Reply:
We don't need a Sentry, it's the attended behaviour.
If the link is protected by a password or not available anymore, we won't have a sharelink, and thus no fileId